doc/guide/go_quickstart.hpp

```diff
@@ -0,0 +1,241 @@
+/**
+ * @file go_quickstart.hpp
+ * @author Yashwant Singh Parihar
+
+@page go_quickstart mlpack in Go quickstart guide
+
+@section go_quickstart_intro Introduction
+
+This page describes how you can quickly get started using mlpack from Go and
+gives a few examples of usage, and pointers to deeper documentation.
+
+This quickstart guide is also available for
+@ref cli_quickstart "the command-line", @ref python_quickstart "Python"
+and @ref julia_quickstart "Julia".
+
+@section go_quickstart_install Installing mlpack
+
+Installing the mlpack bindings for Go is somewhat ticker; you can download
+
+@code{.sh}
+go get -d mlpack.org/v1/mlpack
+make install
```

Comment:
Is the `make install` necessary?

Also, I must be doing something wrong.  I completely removed `~/go`, `~/.cache/go-build`, and `~/.config/go/env` to get a clean Go environment.  Then I ran:

```
go get -d github.com/Yashwants19/v1/
```

but I was surprised to see that mlpack didn't build, and then when I run `go build test.go` I get:

```
$ go build test.go
# github.com/Yashwants19/v1
/usr/bin/ld: cannot find -lmlpack_go_adaboost
/usr/bin/ld: cannot find -lmlpack_go_approx_kfn
/usr/bin/ld: cannot find -lgo_util
/usr/bin/ld: cannot find -lmlpack_go_cf
/usr/bin/ld: cannot find -lgo_util
/usr/bin/ld: cannot find -lmlpack_go_dbscan
/usr/bin/ld: cannot find -lmlpack_go_decision_stump
/usr/bin/ld: cannot find -lmlpack_go_decision_tree
/usr/bin/ld: cannot find -lmlpack_go_det
/usr/bin/ld: cannot find -lmlpack_go_emst
/usr/bin/ld: cannot find -lmlpack_go_fastmks
/usr/bin/ld: cannot find -lmlpack_go_gmm_train
/usr/bin/ld: cannot find -lmlpack_go_hoeffding_tree
/usr/bin/ld: cannot find -lmlpack_go_image_converter
/usr/bin/ld: cannot find -lmlpack_go_kernel_pca
/usr/bin/ld: cannot find -lmlpack_go_kfn
/usr/bin/ld: cannot find -lmlpack_go_kmeans
/usr/bin/ld: cannot find -lmlpack_go_knn
/usr/bin/ld: cannot find -lmlpack_go_krann
/usr/bin/ld: cannot find -lmlpack_go_lars
/usr/bin/ld: cannot find -lmlpack_go_linear_regression
/usr/bin/ld: cannot find -lmlpack_go_linear_svm
/usr/bin/ld: cannot find -lmlpack_go_lmnn
/usr/bin/ld: cannot find -lmlpack_go_local_coordinate_coding
/usr/bin/ld: cannot find -lmlpack_go_logistic_regression
/usr/bin/ld: cannot find -lmlpack_go_lsh
/usr/bin/ld: cannot find -lmlpack_go_mean_shift
/usr/bin/ld: cannot find -lmlpack_go_nbc
/usr/bin/ld: cannot find -lmlpack_go_nca
/usr/bin/ld: cannot find -lmlpack_go_nmf
/usr/bin/ld: cannot find -lmlpack_go_pca
/usr/bin/ld: cannot find -lmlpack_go_perceptron
/usr/bin/ld: cannot find -lmlpack_go_preprocess_binarize
/usr/bin/ld: cannot find -lmlpack_go_preprocess_describe
/usr/bin/ld: cannot find -lmlpack_go_preprocess_scale
/usr/bin/ld: cannot find -lmlpack_go_preprocess_split
/usr/bin/ld: cannot find -lmlpack_go_radical
/usr/bin/ld: cannot find -lmlpack_go_random_forest
/usr/bin/ld: cannot find -lmlpack_go_range_search
/usr/bin/ld: cannot find -lmlpack_go_softmax_regression
/usr/bin/ld: cannot find -lmlpack_go_sparse_coding
/usr/bin/ld: cannot find -lmlpack_go_test_go_binding
collect2: error: ld returned 1 exit status
```

Did I mess something up?  Let me know what I should do to fix it.  (I am definitely not a Go expert...)

Reply:
Have you `make install` the package ?

`go get -d github.com/Yashwants19/v1/` only downloading the package and not installing it.
After downloading we have to build mlpack and other go shared libraries using `make install`.

Reply:
There's no `Makefile` in the directory, though.  Here's a transcript:

```
$ cd go-test
$ go get -d github.com/Yashwants19/v1/
$ ls
test.go
$ make
make: *** No targets specified and no makefile found.  Stop.
$ make install
make: *** No rule to make target 'install'.  Stop.
```

I'm not sure what's different about this than what I did last time I tested it.

Reply:
I am really sorry we should change the directory too. 
I will update this.
`cd ${GOPATH}/src/github.com/Yashwants19/v1`

Reply:
Sorry forget to update `Yashwants19/v1` too, Earlier you may not able to use new utility functions. 
Just pushed/update the changes. 
Thank you.

Reply:
Awesome, thank you!  Changing to the directory go downloaded the source to makes it work perfectly. :+1: